tests/conftest.py

```diff
@@ -281,6 +267,30 @@ def _fc(request, fc_module, new_user=False):
     return fc_module
 
 
+def _port(port, busy_ports, ip=_LOCALHOST):
+    import socket
+    from pykern.pkdebug import pkdlog
+
+    p = int(port)
+    while sirepo.const.PORT_MIN <= p <= sirepo.const.PORT_MAX:
+        with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+            s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+            try:
+                assert p not in busy_ports
+                s.bind(
+                    (
+                        ip,
+                        p,
+                    )
+                )
+                busy_ports.append(p)
+                return str(p)
+            except Exception:
+                pkdlog(f"ip={ip} port={p} in use")
+            p += 1
+    raise AssertionError(f"ip={ip} port={p} invalid address")
```

Comment:
error is "unable to allocate port".

Reply:
[52e56a4](https://github.com/radiasoft/sirepo/pull/4846/commits/52e56a438b57f19b8e689580bc4ada1cdc76541f)